### PR TITLE
DICOM Overlays

### PR DESCRIPTION
**Pull request by kofifus**
Adds 'overlayPlaneModule' type returning an array of DICOM overlays.
This array is then added the 'overlays' propertry of the image.

This is then read by the overlays tool see - https://github.com/cornerstonejs/cornerstoneTools/pull/788

see issue https://github.com/cornerstonejs/cornerstoneTools/issues/780
see DICOM overlays standard - http://dicom.nema.org/dicom/2013/output/chtml/part03/sect_C.9.html

### CODE REVIEW

File: src/imageLoader/wadouri/metaData/metaDataProvider.js [hunk 1]
```diff
@@ -121,6 +121,44 @@ function metaDataProvider (type, imageId) {
       }
     };
   }
+
+  if (type === 'overlays') {
```
Comment (dannyrb) [inline src/imageLoader/wadouri/metaData/metaDataProvider.js]: To be consistent w/ the other modules in this `metaDataLoader`, this should probably be `overlayPlaneModule`

Comment (kofifus) [reply src/imageLoader/wadouri/metaData/metaDataProvider.js]: done, also I comitted the change to createImage I forgot !

File: src/imageLoader/createImage.js [hunk 2]
```diff
@@ -186,6 +186,13 @@ function createImage (imageId, pixelData, transferSyntax, options) {
         image.windowWidth = maxVoi - minVoi;
         image.windowCenter = (maxVoi + minVoi) / 2;
       }
+
+      // overlays
+      const overlays = cornerstone.metaData.get('overlayPlaneModule', imageId);
+      if (overlays && overlays.length > 0) {
+        image.overlays = overlays;
+      }
+
```
Comment (dannyrb) [inline src/imageLoader/createImage.js]: If we leave this off of `createImage` for now (we can discuss later), then I feel comfortable merging this after a quick gut check from @swederik. This should give us just enough to test an `OverlayTool` -- we can iterate and clean this and the corresponding tool up after we get an example and working proof-of-concept.

Comment (kofifus) [reply src/imageLoader/createImage.js]: if you leave it off createImage how is it going to be available for the tool ?

Comment (dannyrb) [reply src/imageLoader/createImage.js]: Tool's can grab metadata by asking registered metadata providers:

`cornerstone.metaData.get('overlayPlaneModule', imageId);`

You can see an example of us doing it in the [current length tool src](https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/tools/annotation/LengthTool.js#L126-L129).

Comment (kofifus) [reply src/imageLoader/createImage.js]: got it ..

Comment (kofifus) [reply src/imageLoader/createImage.js]: removed createimage.js from the PR

Comment (kofifus) [reply src/imageLoader/createImage.js]: yeah that's what I understood
so I removed createimage.js changes from the PR - they are not needed
and added changes to your tool to read directly from metadata

**Comment by swederik** [comment]
Thanks! This is a frequently requested feature so I am sure the community will be happy to have it.

It looks like the PR actually deletes the createImage file entirely though. Could you add that back in?

This will add some overlay data for WADO-URI but doesn't address WADO-RS. I'm not sure how we want to handle that, since I think with WADO-RS we will have to retrieve it from the server via a separate bulk data URI (similar to the palette color lookup tables). That can be a separate step though.

One comment: I don't see the reason why this provider should include

```
fillStyle: 'yellow',
visible: true
```

since those aren't pulled from the metadata. That should be set in the tool instead.

**Pull request closed without merging**